app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/AccountCheckController.scala

```diff
@@ -57,31 +65,34 @@ class AccountCheckController @Inject()(
 
   def accountCheck(redirectUrl: String): Action[AnyContent] = authAction.async {
     implicit request =>
-      sessionCache.save[String]("redirectURL", redirectUrl)
+      sessionCache.save[String](REDIRECT_URL, redirectUrl)
       accountCheckOrchestrator.getAccountType.value.flatMap {
-        case Right(SINGLE_ACCOUNT) => silentEnrol()
         case Right(PT_ASSIGNED_TO_CURRENT_USER) =>
           Future.successful(Redirect(redirectUrl))
         case Right(PT_ASSIGNED_TO_OTHER_USER) =>
           Future.successful(
             Redirect(routes.PTEnrolmentOnOtherAccountController.view)
           )
-        case Right(MULTIPLE_ACCOUNTS) =>
-          Future.successful(Redirect(routes.LandingPageController.view))
-        case Left(_) => Future.successful(InternalServerError)
+        case Right(accountType) => silentEnrolmentAndRedirect(accountType)
+        case Left(_)            => Future.successful(InternalServerError)
       }
   }
 
-  private def silentEnrol()(
+  private def silentEnrolmentAndRedirect(accountType: AccountTypes.Value)(
```

Comment:
Hasn't this changed behaviour? In the MULTIPLE_ACCOUNTS journey, the user is now being enrolled on the landing page when they weren't before.

Reply:
In the multiple accounts journey (no SA or SA on signed in credential) a forced enrolment is occurring and then the LandingPage is displayed to the user. Ideally, as an action is happening, it would be a POST on the previous page however, we don't have that so the next best thing is to do it just before the redirect. Any refreshing of the page etc will then not cause an action to be repeated.